grafana/grafana
# Elasticsearch: Fix legend generated on backend

This PR fixes logic when creating legend on backend with 2 changes:

1. We are moving name for series from `frame.name` to `field.Config.DisplayNameFromDS`. This is basically reverting https://github.com/grafana/grafana/pull/64233 that we've done to make sure that name for frames created on frontend and backend is at the same place. Unfortunately, this was creating issue because on backend we are (correctly) adding labels to frame which resulted in Grafana attaching additional `labels` behind the series name. This wasn't issue on frontend as we weren't adding `labels` to frames.

2. On backend, we are ensuring that `metric.Field` (which has a metric name) is appended and not `metric.ID`. 

How to test:

1. Run `make devenv sources=elastic`
2. Run `cd devenv` and then `./setup.sh`
3. Go to `http://localhost:3000/d/es_migration/datasource-tests-elasticsearch-migration?orgId=1` and make sure that legends on frontend running queries and backend running queries are same. For now, skip `Multiple metrics and aggregations` section where legend is not the same. This is due to the fact, that field has name `metric` and we use in backend `metric` as label where we store metric name. Fix for this will come in separate PR. 

Part of #54011
